src/error.rs

```diff
@@ -42,6 +41,8 @@ pub enum Error {
     BadResponse(String),
     #[error("timeout expired")]
     TimedOut,
+    #[error("error while parsing columns header from the response: {0}")]
+    InvalidColumnsHeader(#[source] BoxedError),
```

Comment:
This error can include `TypesError::NotEnoughData`, which is basically `Error::NotEnoughData`, often with the same underlying root cause.